docs/design/fabric-v2+/fpc-key-dist.puml

```diff
@@ -213,111 +181,135 @@ group key distribution
   return true /'Enclave2 -> TLCC_Enclave2'/
-    note left
-        if false, abort
-    end note
-  Enclave2      -> ERCC2            ++: GetRegistration(ECC1_S_PK)
-  return registration_message <- <CC_name, CC_ver, CH_id, org_id, MSP_id_org, SIG_org, ECC1_S_PK, ECC1_E_PK, attestation_report> /'Enclave2 -> ERCC2'/
-  Enclave2      -> TLCC_Enclave2    ++: check_committed_data(registration_message)
+  note left
+    if false, abort
+  end note
+  Enclave2      -> Enclave2         :   extract Enclave1_VK from cckeydist_start_message.key_distribution_message
+  Enclave2      -> ERCC2            ++: GetRegistration(Enclave1_VK)
+  return Registered_Enclave1_Credentials /'Enclave2 -> ERCC2'/
+  Enclave2      -> TLCC_Enclave2    ++: check_committed_data(Registered_Enclave1_Credentials)
   return true /'Enclave2 -> TLCC_Enclave2'/
-    note left
-        if false, abort
-    end note
-  Enclave2      -> Enclave2         :   verify attestation report
-  Enclave2      -> Enclave2         :   verify ECC1_S_PK matches in registration_message and keydist_start_message
-  Enclave2      -> Enclave2         :   verify SIG_ECC1
-  Enclave2      -> Enclave2         :   SEK, CC_E_SK <- decrypt(keydist_start_message.encrypted_keys_message) with ECC2_E_SK
-
-  Enclave2      -> Enclave2         :   SIG_ECC2 <- sign <CC_E_PK> with ECC2_S_SK
-  Enclave2      -> Enclave2         :   store SEK, CC_E_SK in sealed_state
-  Enclave2      -> Enclave2         :   key_confirmation_message <- <ECC2_S_PK, SIG_ECC2, CC_E_PK>
-
-  return key_confirmation_message, sealed_state /'Enclave2 -> ECC2'/
-  return key_confirmation_message, sealed_state /'ECC2 -> ECC_Manager2'/
+  note left
+    if false, abort
+  end note
+  Enclave2      -> Enclave2         :   check Enclave1_VK matches in registration_message and cckeydist_start_message
+  Enclave2      -> Enclave2         :   check CC_Id in Registered_Enclave1_Credentials matches internal one
+  note over Enclave2
+    Since Enclave1's Credentials are committed, then Enclave1's Evidence has already been verified by ERCC.
+    Hence, verifying Enclave1's Evidence at this point does not appear necessary.
+  end note
+  Enclave2      -> Enclave2         :   verify Sig_Enclave1
+  Enclave2      -> Enclave2         :   SEK, Chaincode_DK <- decrypt(cckeydist_start_message.encrypted_keys_message) with Enclave2_DK
+  Enclave2      -> Enclave2         :   extract Chaincode_EK from Chaincode_DK
+  Enclave2      -> Enclave2         :   Sig_Enclave2 <- sign Chaincode_EK with Enclave2_SK
+  Enclave2      -> Enclave2         :   cckey_registration_message <- <Enclave2_VK, Sig_Enclave2, Chaincode_EK>
 
-  ECC_Manager2  -> Storage2     :   store sealed_state
+  return cckey_registration_message, /'Enclave2 -> ECC2'/
+  return cckey_registration_message, /'ECC2 -> E_Manager2'/
 
-  return key_confirmation_message /'Admin2 -> ECC_Manager2'/
+  return cckey_registration_message /'Admin2 -> E_Manager2'/
```

Comment:
If we use seal above, we should do here also.

Reply:
added comment